litestar/typing.py

```diff
@@ -529,3 +557,14 @@ def from_parameter(cls, parameter: Parameter, fn_type_hints: dict[str, Any]) ->
             name=parameter.name,
             default=Empty if parameter.default is Signature.empty else parameter.default,
         )
+
+    def test_predicate_recursively(self, predicate: Callable[[FieldDefinition], bool]) -> bool:
```

Comment:
```suggestion
    def matches_predicate(self, predicate: Callable[[FieldDefinition], bool]) -> bool:
```

`test_` doesn't really match the naming convention of the rest of the methods.

Reply:
`matches_predicate_recursively`? i want to make it clear there is recursion involved here.